scripts/build.mjs

```diff
@@ -143,6 +152,11 @@ yargs(process.argv.slice(2))
           describe:
             "Set to `true` if you don't want to generate a package.json file in the /esm folder.",
         })
+        .option('cjsDir', {
```

Comment:
Does it make sense to do something different in base than in core?

Reply:
Currently, base has two outputs `cjs` and `esm` dirs unlike core, were cjs is at the top level of build.
If we want to change the structure in either of the repos, it makes sense to do it in base since it is in alpha anyways.
Though, IMO, explicit `cjs` and `esm` (and `modern`) seems cleaner to me than what we are doing in core currently.

Reply:
The idea behind the structure in core was to keep support for node10 resolution. Not sure how you can achieve that with the base structure.
The current structure makes it too easy to forget properties like `sideEffects` in the subfolder package.json (as is currently hapening in base)
IMO the ideal structure is a flat one with cjs and mjs file extensions. This avoids package.json lookups to determine module type. But that's a bit more difficult to achieve with our current setup.

Reply:
<img width="937" alt="Screenshot 2025-04-02 at 11 02 52 AM" src="https://github.com/user-attachments/assets/0b9a490d-2dae-4d14-98c1-de7084bf0e33" />

I was surprised to find that this structure [supports](https://app.circleci.com/pipelines/github/mui/base-ui/8704/workflows/b4474813-e242-4986-9ab3-288f20114fd3/jobs/81777/steps) node10 resolution.

Reply:
Something different in `base-ui`'s package.json is that it list each export individually. There are no `*` exports.
So if supporting `node10` resolution was the only issue for us, I am sure we can restructure core's build output.

Reply:
> The current structure makes it too easy to forget properties like sideEffects in the subfolder package.json (as is currently hapening in base)

But with the updated code to use latest infra changes, this has been corrected.

Reply:
> I was surprised to find that this structure supports node10 resolution.

The type resolution yes, through `typesVersions`, I'm not sure it would actually resolve the javascript correctly under node10, I don't see how without exports field support.

Reply:
> Something different in base-ui's package.json is that it lists each export individually

Yes, we're likely also going to move to something like that eventually. There's a `mui-src` condition that is supported by our package.json exports generator that helps with this. e.g. https://github.com/mui/material-ui/blob/7a7c530e07ae959baf1ca660053cc42da2187bbd/packages/mui-material/package.json#L109

Reply:
@Janpot Yeah. The node10 resolution only works for types. Not for the implementation file.